src/devices/Tlc1543/Tlc1543.cs

```diff
@@ -0,0 +1,239 @@
+        /// <remarks>
+        /// <br>Refer to table 2 and 3 in TLC1543 documentation for more information</br>
+        /// </remarks>
+        /// </summary>
+        public Channel ChargeChannel
+        {
+            get => _chargeChannel;
+            set
+            {
+                if ((int)value >= 0 & (int)value < 14)
+                {
+                    _chargeChannel = value;
+                }
+                else
+                {
+                    throw new ArgumentOutOfRangeException("ChargeChannel", "Out of range. \nMust be non-negative and less than 14. \nRefer to table 2 and 3 in TLC1543 documentation for more information");
+                }
+            }
+        }
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Constructor for Tlc1543
+        /// </summary>
+        /// <param name="address">Address</param>
+        /// <param name="chipSelect">Chip Select</param>
+        /// <param name="dataOut">Data Out</param>
+        /// <param name="inputOutputClock">I/O Clock</param>
+        /// <param name="controller"> s</param>
+        /// <param name="pinNumberingScheme"> s</param>
+        /// <param name="shouldDispose"> s</param>
+        public Tlc1543(int address, int chipSelect, int dataOut, int inputOutputClock, GpioController controller = null, PinNumberingScheme pinNumberingScheme = PinNumberingScheme.Logical, bool shouldDispose = true)
+        {
+            _shouldDispose = controller == null || shouldDispose;
+            _digital = controller ?? new GpioController(pinNumberingScheme);
+            _address = address;
+            _chipSelect = chipSelect;
+            _dataOut = dataOut;
+            _inputOutputClock = inputOutputClock;
+
+            _digital.OpenPin(_address, PinMode.Output);
+            _digital.OpenPin(_chipSelect, PinMode.Output);
+            _digital.OpenPin(_dataOut, PinMode.InputPullUp);
+            _digital.OpenPin(_inputOutputClock, PinMode.Output);
+        }
+
+        /// <summary>
+        /// Constructor for Tlc1543
+        /// </summary>
+        /// <param name="address">Address</param>
+        /// <param name="chipSelect">Chip Select</param>
+        /// <param name="dataOut">Data Out</param>
+        /// <param name="endOfConversion">End of Conversion</param>
+        /// <param name="inputOutputClock">I/O Clock</param>
+        public Tlc1543(int address, int chipSelect, int dataOut, int inputOutputClock, int endOfConversion)
+        {
+            _address = address;
+            _chipSelect = chipSelect;
+            _dataOut = dataOut;
+            _endOfConversion = endOfConversion;
+            _inputOutputClock = inputOutputClock;
+
+            _digital.OpenPin(_address, PinMode.Output);
+            _digital.OpenPin(_chipSelect, PinMode.Output);
+            _digital.OpenPin(_dataOut, PinMode.InputPullUp);
+            _digital.OpenPin((int)_endOfConversion, PinMode.InputPullUp);

```

Comment:
why does this need cast? should this support passing -1 if this is not supported?